test/integration/player_is_told_to_try_again_if_they_reenter_a_move_test.exs

```diff
@@ -0,0 +1,116 @@
+defmodule PlayerIsToldToTryAgainIfTheyReenterAMoveTest do
+  use ExUnit.Case
+
+  import Mox
+
+  setup :verify_on_exit!
+
+  test "the player is told to try again if they reenter a move" do
+    randomizer =
+      MockRandomizer
+      |> MockRandomizerHelper.allow_random_coordinate_pair_to_return()
+
+    empty_tile_location = "4E"
+    bomb_tile_location = "0A"
+
+    reader =
+      MockReader
+      |> expect(:read, fn -> empty_tile_location end)
+      |> expect(:read, fn -> empty_tile_location end)
+      |> expect(:read, fn -> bomb_tile_location end)
```

Comment:
Do you have the player lose in order to halt the game execution for the tests? I'm wondering if there's a cleaner way to do this.

Reply:
@Thomascountz yes, that's right. I used to inject `GameRules` and use a mock of it that would return `true` for `player_lost?` when I wanted to force the game to end. I removed that when I added the bombs (and instead forced the user to select a bomb tile) so that the integration tests would be more realistic and and so I could avoid injecting another thing for the sake of the tests. But thinking about it, maybe ending the game by forcing the user to land on a bomb is only really necessary for the integration test that ensures a user loses when they land on a bomb? Maybe I go back to injecting `GameRules` for the other integration tests? Let me know what you think!

Reply:
I hear what you're saying about injecting just for the sake of a test. Phrased another way, I might ask "how does a player exit the game?" I know it wasn't part of this round of user stories, but in the future we might consider how the game will handle "exiting" and "losing"/"winning" as discrete states.

Reply:
Ahh thank you! That's a good point—I suppose I'll leave it as is until we add the ability for a user to exit the game? And then I'll update the integration tests to use that instead